include/misc/civiltime.h

```diff
@@ -0,0 +1,74 @@
+/*
+ * Copyright (c) 2019, Peter Bigot Consulting, LLC
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Utilities for converting between civil time and UNIX time
+ */
+
+#ifndef ZEPHYR_INCLUDE_MISC_CIVILTIME_H_
+#define ZEPHYR_INCLUDE_MISC_CIVILTIME_H_
+
+#include <zephyr/types.h>
+#include <time.h>
+
+/** @brief Convert a civil (proleptic Gregorian) date to days relative to
+ * 1970-01-01.
+ *
+ * @param y the calendar year
+ * @param m the calendar month, in the range [1, 12]
+ * @param d the day of the month, in the range [1, last_day_of_month(y, m)]
+ *
+ * @return the signed number of days between 1970-01-01 and the
+ * specified day.  Negative days are before 1970-01-01.
+ */
+s64_t time_days_from_civil(s64_t y,
```

Comment:
Please use more descriptive names than "y", etc.

Reply:
The variable names match the reference implementation from which this was translated.

Reply:
> reference implementation from which this was translated.

Then it should have licensing and copyright of it, in addition to yours? But even then, as long as you modify it at all, you can use more descriptive names.